<?xml version="1.0" ?>
<robot name="fwomni" xmlns:xacro="http://www.ros.org/wiki/xacro">

<xacro:include filename="$(find fwomni)/urdf/materials.xacro" />
<xacro:include filename="$(find fwomni)/urdf/fwomni.trans" />
<xacro:include filename="$(find fwomni)/urdf/fwomni.gazebo" />

<link name="base_footprint">
</link>

<link name="base_link">
  <inertial>
    <origin xyz="0.0014510929492462577 -1.1820705140781097e-17 0.00527905633638948" rpy="0 0 0"/>
    <mass value="3.019927036453307"/>
    <inertia ixx="0.009026" iyy="0.009305" izz="0.018259" ixy="-0.0" iyz="-0.0" ixz="-5.4e-05"/>
  </inertial>
  <visual>
    <origin xyz="0 0 0.005" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://fwomni/meshes/base_link.stl" scale="0.001 0.001 0.001"/>
    </geometry>
    <material name="silver"/>
  </visual>
  <collision>
    <origin xyz="0 0 0.005" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://fwomni/meshes/base_link.stl" scale="0.001 0.001 0.001"/>
    </geometry>
  </collision>
</link>

<link name="rplidar">
  <inertial>
    <origin xyz="0.0 0.0 0.0030000000000000027" rpy="0 0 0"/>
    <mass value="0.02696735279859846"/>
    <inertia ixx="1e-06" iyy="1e-06" izz="2e-06" ixy="0.0" iyz="0.0" ixz="-0.0"/>
  </inertial>
  <visual>
    <origin xyz="-0.065 -0.0 -0.025" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://fwomni/meshes/rplidar.stl" scale="0.001 0.001 0.001"/>
    </geometry>
    <material name="silver"/>
  </visual>
  <collision>
    <origin xyz="-0.065 -0.0 -0.025" rpy="0 0 0"/>
    <geometry>
      <mesh filename="package://fwomni/meshes/rplidar.stl" scale="0.001 0.001 0.001"/>
    </geometry>
  </collision>
</link>

<link name="front_right">
  <inertial>
    <origin xyz="0.0030000000000000014 -2.220446049250313e-18 1.1779832153568586e-18" rpy="0 0 0"/>
    <mass value="0.01479690139840792"/>
    <inertia ixx="0.0" iyy="1e-06" izz="0.0" ixy="-0.0" iyz="-0.0" ixz="0.0"/>
  </inertial>
  <visual>
    <origin xyz="0 0 0" rpy="0 0 -1.5708"/>
    <geometry>
      <mesh filename="package://fwomni/meshes/wheel.stl" scale="0.001 0.001 0.001"/>
    </geometry>
    <material name="silver"/>
  </visual>
  <collision>
    <origin xyz="0 0 0" rpy="0 0 -1.5708"/>
    <geometry>
      <mesh filename="package://fwomni/meshes/wheel.stl" scale="0.001 0.001 0.001"/>
    </geometry>
  </collision>
</link>

<link name="front_left">
  <inertial>
    <origin xyz="0.0030000000000000014 -2.220446049250313e-18 1.1779832153568586e-18" rpy="0 0 0"/>
    <mass value="0.01479690139840792"/>
    <inertia ixx="0.0" iyy="1e-06" izz="0.0" ixy="-0.0" iyz="-0.0" ixz="0.0"/>
  </inertial>
  <visual>
    <origin xyz="0 0 0" rpy="0 0 -1.5708"/>
    <geometry>
      <mesh filename="package://fwomni/meshes/wheel.stl" scale="0.001 0.001 0.001"/>
    </geometry>
    <material name="silver"/>
  </visual>
  <collision>
    <origin xyz="0 0 0" rpy="0 0 -1.5708"/>
    <geometry>
      <mesh filename="package://fwomni/meshes/wheel.stl" scale="0.001 0.001 0.001"/>
    </geometry>
  </collision>
</link>

<link name="rear_left">
  <inertial>
    <origin xyz="0.0030000000000000014 -2.220446049250313e-18 1.1779832153568586e-18" rpy="0 0 0"/>
    <mass value="0.01479690139840792"/>
    <inertia ixx="0.0" iyy="1e-06" izz="0.0" ixy="-0.0" iyz="-0.0" ixz="0.0"/>
  </inertial>
  <visual>
    <origin xyz="0 0 0" rpy="0 0 -1.5708"/>
    <geometry>
      <mesh filename="package://fwomni/meshes/wheel.stl" scale="0.001 0.001 0.001"/>
    </geometry>
    <material name="silver"/>
  </visual>
  <collision>
    <origin xyz="0 0 0" rpy="0 0 -1.5708"/>
    <geometry>
      <mesh filename="package://fwomni/meshes/wheel.stl" scale="0.001 0.001 0.001"/>
    </geometry>
  </collision>
</link>

<link name="rear_right">
  <inertial>
    <origin xyz="0.0030000000000000014 -2.220446049250313e-18 1.1779832153568586e-18" rpy="0 0 0"/>
    <mass value="0.01479690139840792"/>
    <inertia ixx="0.0" iyy="1e-06" izz="0.0" ixy="-0.0" iyz="-0.0" ixz="0.0"/>
  </inertial>
  <visual>
    <origin xyz="0 0 0" rpy="0 0 -1.5708"/>
    <geometry>
      <mesh filename="package://fwomni/meshes/wheel.stl" scale="0.001 0.001 0.001"/>
    </geometry>
    <material name="silver"/>
  </visual>
  <collision>
    <origin xyz="0 0 0" rpy="0 0 -1.5708"/>
    <geometry>
      <mesh filename="package://fwomni/meshes/wheel.stl" scale="0.001 0.001 0.001"/>
    </geometry>
  </collision>
</link>

<joint name="Rigid1" type="fixed">
  <origin xyz="0 0 0" rpy="0 0 0"/>
  <parent link="base_footprint"/>
  <child link="base_link"/>
</joint>

<joint name="Rigid2" type="fixed">
  <origin xyz="0.065 0.0 0.03" rpy="0 0 0"/>
  <parent link="base_link"/>
  <child link="rplidar"/>
</joint>

<joint name="Cont1" type="continuous">
  <origin xyz="0.085328 -0.085328 0.01" rpy="0 0 0.785398"/>
  <parent link="base_link"/>
  <child link="front_right"/>
  <axis xyz="0 1 0"/>
</joint>

<joint name="Cont2" type="continuous">
  <origin xyz="0.085328 0.085328 0.01" rpy="0 0 2.35619"/>
  <parent link="base_link"/>
  <child link="front_left"/>
  <axis xyz="0 1 0"/>
</joint>

<joint name="Cont3" type="continuous">
  <origin xyz="-0.085328 0.085328 0.01" rpy="0 0 -2.35619"/>
  <parent link="base_link"/>
  <child link="rear_left"/>
  <axis xyz="0 1 0"/>
</joint>

<joint name="Cont4" type="continuous">
  <origin xyz="-0.085328 -0.085328 0.01" rpy="0 0 -0.785398"/>
  <parent link="base_link"/>
  <child link="rear_right"/>
  <axis xyz="0 1 0"/>
</joint>

</robot>
